src/config/Config.ts

```diff
@@ -15,34 +15,113 @@
  */
 
 import {TopicOverloadPolicy} from '../proxy/topic/TopicOverloadPolicy';
-import {ClientNetworkConfig} from './ClientNetworkConfig';
+import {ClientNetworkConfig, ClientNetworkConfigImpl} from './ClientNetworkConfig';
 import {ConfigPatternMatcher} from './ConfigPatternMatcher';
 import {EvictionPolicy} from './EvictionPolicy';
-import {FlakeIdGeneratorConfig} from './FlakeIdGeneratorConfig';
-import {ImportConfig, ListenerImportConfig} from './ImportConfig';
+import {FlakeIdGeneratorConfig, FlakeIdGeneratorConfigImpl} from './FlakeIdGeneratorConfig';
 import {InMemoryFormat} from './InMemoryFormat';
-import {ListenerConfig} from './ListenerConfig';
-import {NearCacheConfig} from './NearCacheConfig';
+import {MembershipListener} from '../core/MembershipListener';
+import {LifecycleState} from '../LifecycleService';
+import {NearCacheConfig, NearCacheConfigImpl} from './NearCacheConfig';
 import {SSLConfig} from './SSLConfig';
 import {Properties} from './Properties';
-import {ReliableTopicConfig} from './ReliableTopicConfig';
-import {SerializationConfig} from './SerializationConfig';
+import {ReliableTopicConfig, ReliableTopicConfigImpl} from './ReliableTopicConfig';
+import {SerializationConfig, SerializationConfigImpl} from './SerializationConfig';
 import {Statistics} from '../statistics/Statistics';
 import {LogLevel} from '..';
 import {ILogger} from '../logging/ILogger';
 import {JsonStringDeserializationPolicy} from './JsonStringDeserializationPolicy';
-import {ConnectionStrategyConfig, ReconnectMode} from './ConnectionStrategyConfig';
-import {LoadBalancer} from '../LoadBalancer';
+import {ConnectionStrategyConfig, ConnectionStrategyConfigImpl, ReconnectMode} from './ConnectionStrategyConfig';
+import {LoadBalancerType, LoadBalancerConfig, LoadBalancerConfigImpl} from './LoadBalancerConfig';
 import {IndexConfig} from './IndexConfig';
 import {IndexType} from './IndexType';
 import {ConnectionRetryConfig} from './ConnectionRetryConfig';
 
-const DEFAULT_CLUSTER_NAME = 'dev';
-
 /**
- * Top level configuration object of Hazelcast client. Other configurations items are properties of this object.
+ * Top level configuration object of Hazelcast client.
+ * Other configurations items are properties of this object.
  */
-export class ClientConfig {
+export interface ClientConfig {
+
+    /**
+     * Name of the cluster to connect to. By default, set to `dev`.
+     */
+    clusterName?: string;
+
+    /**
+     * Name of the client instance. By default, set to `hz.client_${CLIENT_ID}`, where
+     * `CLIENT_ID` starts from `0` and it is incremented by `1` for each new client.
+     */
+    instanceName?: string;
```

Comment:
What do you think about changing this to `clientName`. I believe the latter one is more intuitive but it is different from the Java counterpart.

Reply:
I think we should try to keep naming close enough to Java client when possible. So, I'd rather keep it as is.